src/api/content-rest-api/api/categories.api.ts

```diff
@@ -0,0 +1,496 @@
+
+        const queryParams = {};
+        const headerParams = {};
+        const formParams = {};
+
+        const contentTypes = ['application/json'];
+        const accepts = ['application/json'];
+
+        return this.apiClient.callApi(
+            '/category/{categoryId}', 'DELETE',
+            pathParams, queryParams, headerParams, formParams, postBody,
+            contentTypes, accepts );
+    }
+
+    /**
+    * Unassign a node from category
+    *
+    * Removes the node **nodeId** from the category **categoryId**.
+    *
+    * @param nodeId The identifier of a node.
+    * @param categoryId The identifier of a category.
+    * @return Promise<{}>
+    */
+     unlinkNodeFromCategory(nodeId: string, categoryId: string): Promise<any> {
+        throwIfNotDefined(nodeId, 'nodeId');
+        throwIfNotDefined(categoryId, 'categoryId');
+        const postBody: null = null;
+
+        const pathParams = {
+            'nodeId': nodeId,
+            'categoryId': categoryId
+        };
+
+        const queryParams = {};
+        const headerParams = {};
+        const formParams = {};
+
+        const contentTypes = ['application/json'];
+        const accepts = ['application/json'];
+
+        return this.apiClient.callApi(
+            '/nodes/{nodeId}/category-links/{categoryId}', 'DELETE',
+            pathParams, queryParams, headerParams, formParams, postBody,
+            contentTypes, accepts );
+    }
+
+    /**
+    * Update a category
+    *
+    * Updates the category **categoryId**.
+    * You must have admin rights to update a category.
+    *
+    * @param categoryId The identifier of a category.
+    * @param categoryBodyUpdate The updated category
+    * @param opts Optional parameters
+    * @param opts.fields A list of field names.
+
+    You can use this parameter to restrict the fields
+    returned within a response if, for example, you want to save on overall bandwidth.
+
+    The list applies to a returned individual
+    entity or entries within a collection.
+
+    If the API method also supports the **include**
+    parameter, then the fields specified in the **include**
+    parameter are returned in addition to those specified in the **fields** parameter.
+
+    * @return Promise<CategoryEntry>
+    */
+    updateCategory(categoryId: string, categoryBodyUpdate: CategoryBody, opts?: any): Promise<CategoryEntry> {
+
+        throwIfNotDefined(categoryId, 'categoryId');
+        throwIfNotDefined(categoryBodyUpdate, 'categoryBodyUpdate');
+
+        opts = opts || {};
+        const postBody = categoryBodyUpdate;
+
+        const pathParams = {
+            'categoryId': categoryId
+        };
+
+        const queryParams = {
+            'fields': buildCollectionParam(opts['fields'], 'csv')
+        };
+
+        const headerParams = {};
+        const formParams = {};
+
+        const contentTypes = ['application/json'];
+        const accepts = ['application/json'];
+
+        return this.apiClient.callApi(
```

Comment:
Please see http-client.interface.ts, I think we should use function slike put, post etc. because callApi is deprecated but these not. Here is copy of that file: 
```ts
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Authentication } from '../authentication/authentication';
import { Emitter } from 'event-emitter';

export interface RequestOptions {
    path: string;
    httpMethod?: string;
    pathParams?: any;
    queryParams?: any;
    headerParams?: any;
    formParams?: any;
    bodyParam?: any;
    contentTypes?: string[];
    accepts?: string[];
    returnType?: any;
    contextRoot?: string;
    responseType?: string;
    url?: string;
    readonly accept?: string;
    readonly contentType?: string;
}

export interface HttpClientConfig {
    contextRoot: string;
    host?: string; // Should be mandatory but can't make it because of AlfrescoApiConfig incompatibility 😕
    servicePath?: string; // Should be mandatory but can't make it because of AlfrescoApiConfig incompatibility 😕
}

export interface LegacyHttpClient {
    basePath: string;
    config: HttpClientConfig;

    request<T = any>(options: RequestOptions): Promise<T>;
    post<T = any>(options: RequestOptions): Promise<T>;
    put<T = any>(options: RequestOptions): Promise<T>;
    get<T = any>(options: RequestOptions): Promise<T>;
    delete<T = void>(options: RequestOptions): Promise<T>;
    /** @deprecated */
    callApi(
        path: string,
        httpMethod: string,
        pathParams?: any,
        queryParams?: any,
        headerParams?: any,
        formParams?: any,
        bodyParam?: any,
        contentTypes?: string[],
        accepts?: string[],
        returnType?: any,
        contextRoot?: string,
        responseType?: string,
        url?: string
    ): Promise<any>;
    /** @deprecated */
    callCustomApi(
        path: string,
        httpMethod: string,
        pathParams?: any,
        queryParams?: any,
        headerParams?: any,
        formParams?: any,
        bodyParam?: any,
        contentTypes?: string[],
        accepts?: string[],
        returnType?: any,
        contextRoot?: string,
        responseType?: string
    ): Promise<any>;
}

export interface SecurityOptions {
    readonly isBpmRequest: boolean;
    readonly enableCsrf?: boolean;
    readonly withCredentials?: boolean;
    readonly authentications: Authentication;
    readonly defaultHeaders: Record<string, string>;
}

export interface Emitters {
    readonly eventEmitter: Emitter;
    readonly apiClientEmitter: Emitter;
}

export interface HttpClient {
    request<T = any>(url: string, options: RequestOptions, security: SecurityOptions, emitters: Emitters): Promise<T>;
    post<T = any>(url: string, options: RequestOptions, security: SecurityOptions, emitters: Emitters): Promise<T>;
    put<T = any>(url: string, options: RequestOptions, security: SecurityOptions, emitters: Emitters): Promise<T>;
    get<T = any>(url: string, options: RequestOptions, security: SecurityOptions, emitters: Emitters): Promise<T>;
    delete<T = void>(url: string, options: RequestOptions, security: SecurityOptions, emitters: Emitters): Promise<T>;
}
```

Reply:
I asked @DenysVuika about this and we agreed for now to stick to what other apis are using right now

Reply:
Also if possible we should take these functions from HttpClient rather than LegacyHttpClient (because that is "legacy")